packages/@glimmer/syntax/lib/symbol-table.ts

```diff
@@ -86,7 +87,9 @@ export class ProgramSymbolTable extends SymbolTable {
   }
 
   allocateFree(name: string, resolution: ASTv2.FreeVarResolution): number {
-    if (resolution.resolution() === SexpOpcodes.GetFreeAsComponentHead) {
+    // If the name in question is an uppercase (i.e. angle-bracket) component invocation, run
+    // the optional `customizeComponentName` function provided to the precompiler.
+    if (resolution.resolution() === SexpOpcodes.GetFreeAsComponentHead && isUpperCase(name)) {
```

Comment:
We chatted a bit about this in discord, but I think even after this change we still have an issue. I would expect (for example) that `{{#Foo}}{{/Foo}}` should _also_ not call `customizeComponentName`, prior to #1253 we _only_ attempted transformation for `element.tag` specifically.